tracer/test/test-applications/integrations/Samples.OpenTelemetrySdk/Program.cs

```diff
@@ -165,6 +165,9 @@ private static void RunSpanUpdateMethods(TelemetrySpan span)
 
         using (var innerSpan = _tracer.StartActiveSpan("InnerSpan"))
         {
+            // if we don't set the span as an error we won't check it for exception event
+            // so if we only do RecordException, we wouldn't copy that info over as the span isn't marked as an Error
+            innerSpan.SetStatus(Status.Error.WithDescription("Something went wrong"));
```

Comment:
I think the code comment could also reference this readme in the OTEL .NET docs that says the same: https://github.com/open-telemetry/opentelemetry-dotnet/tree/2916b2de80522d4b1cafe353b3fda3fd629ddb00/docs/trace/reporting-exceptions#option-4---use-activityrecordexception

Reply:
I agree we should add that readme, but I wonder whether we _should_ read those attributes either way 🤔 It doesn't seem to specify that you _must_ set the status if you record an exception?

Reply:
Thanks, added the link to that readme

> I agree we should add that readme, but I wonder whether we _should_ read those attributes either way 🤔 It doesn't seem to specify that you _must_ set the status if you record an exception?

I essentially followed the code in the agent here: https://github.com/DataDog/datadog-agent/blob/67c353cff1a6a275d7ce40059aad30fc6a3a0bc1/pkg/trace/api/otlp.go#L583
and to my understanding (I'm not super familiar with Go) they would only extract the attributes if it is set as an error as well.

Just to see what would happen I'll try it out locally without setting the `Activity`/`Span` as an error but record and extract those attributes regardless.

Reply:
So if it isn't marked as an error, but has that exception event and we read/extract it into tags it'll show up on the "Error" tab, but in the flame graph it won't be red and it won't have that "Error detected" notice.

I think my understanding is that `RecordException` is purely to just record an exception and not necessarily mark the span as an error unlike `ISpan.SetException` which does both. If that is the case then maybe it does make more sense to always check for the `"exception"` attribute as you _can_ call `RecordException` without any requirement for setting it as an error - we'd miss those with this implementation at the moment.

I could see a usecase where an exception is handled, `RecordException` is used to record the exception that occurred, but isn't necessarily an "error" because it was able to be corrected.

Reply:
Thanks for checking that! Yeah, I'm undecided to be honest, for the reasons you provided. The main reason against it IMO is the perf impact of always needing to check for them. Let's see what @zacharycmontoya thinks! 🙂

Reply:
I'm so sorry, I didn't see this until now. Do you still need my input here?